### PR TITLE
CI/CD Tests & Type Checking

### PR DESCRIPTION
**Pull request by JacobMGEvans**
- Testing script
- Lint script
- tsc is using skipLibCheck as a current workaround

### CODE REVIEW

File: .github/workflows/tests-typecheck.yml [hunk 1]
```diff
@@ -0,0 +1,63 @@
+name: Tests, Linter & Typecheck
+
+on:  
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
```
Comment (threepointone) [inline .github/workflows/tests-typecheck.yml:9]: I think we should run this on all pull requests, not just the ones to `main` (It'll be most commonly used with main of course, but we should have it on for all anyway)

**Review by threepointone — CHANGES_REQUESTED** [review]

File: .github/workflows/tests-typecheck.yml (hunk 1 above)
Comment (JacobMGEvans) [reply .github/workflows/tests-typecheck.yml:9]: 100%

File: .github/workflows/tests-typecheck.yml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+        # Prefer offline will skip registry verification with cached NPM packages
+      - name: Install NPM dependencies
+        run: npm ci --prefer-offline
+      
+      - name: Run Type Checking & ESLint
+        #TODO remove the --skipLibCheck flag at some point
+        run: tsc --skipLibCheck && npm run lint
```
Comment (jbergstroem) [inline .github/workflows/tests-typecheck.yml:59]: Perhaps add `--noEmit`?

**Review by threepointone — APPROVED** [review]
Looks good to me!! Merging this baby

**Pull request merged**